include/libtorrent/alert_types.hpp

```diff
@@ -2507,12 +2507,30 @@ namespace libtorrent
 #endif // TORRENT_DISABLE_LOGGING
 	};
 
+	// This alert is generated when a piece has passed the hash check and has
+	// been written to disk (so it can be safely read). This alert may be generated
+	// more than once for each piece.
+	struct TORRENT_EXPORT piece_ready_alert TORRENT_FINAL : torrent_alert
+	{
+		// internal
+		piece_ready_alert(aux::stack_allocator& alloc, torrent_handle h
+			, int piece_num);
+
+		TORRENT_DEFINE_ALERT(piece_ready_alert, 90)
+
+		static const int static_category = alert::progress_notification;
```

Comment:
I would suggest inventing a new alert category, say, ``alert::piece_progress_notification`` and maybe raise the priority of this alert as well (as one may imagine it's important if this category is enabled)

Reply:
Will do.